daemon/algod/api/server/v2/handlers.go

```diff
@@ -69,6 +70,19 @@ const MaxTealSourceBytes = 200_000
 // become quite large, so we allow up to 1MB
 const MaxTealDryrunBytes = 1_000_000
 
+// MaxAssetResults sets a size limit for the number of assets returned in a single request to the
+// /v2/accounts/{address}/assets endpoint
+const MaxAssetResults = 10000
```

Comment:
How do we decide what the appropriate size here is? (1.2k definitely performs fast locally) I'm fine with a batch of 1k for example.

Reply:
I can't decide how to decide this.  I guess we can calculate the maximum size of the response and see what we're comfortable with.  I think each asset is at most something like 1k. (once you consider all the fields in params that can be big, and get json key names and b64 encoded).
Of course, I hope we're gzipping.

I think the main disadvantage to a really high number is that it may encourage lazy usage: "Ehh, I'll grab up to 10,000. No need to worry about pagination."

Reply:
Can we all live with 1k, or want something smaller?

Reply:
When working with the NFT community, I learned that it is not uncommon for NFT power-users to have >1k assets, so it would be great to allow apps to grab up to 1k in a single call and minimize pagination in practice.

Reply:
I'm going with 1000 max and as the default here.